Neos.Fusion/Classes/Core/FusionSourceCode.php

```diff
@@ -0,0 +1,62 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Fusion\Core;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Fusion;
+
+#[Flow\Proxy(false)]
+class FusionSourceCode
+{
+    private function __construct(
+        private string $sourceCode,
+        private ?string $contextPathAndFilename
+    ) {
+        if (trim($sourceCode) === '') {
+            throw new Fusion\Exception("The sourcecode of the FusionSourceCode must not be empty. $this->contextPathAndFilename", 1657963664);
+        }
```

Comment:
We cant do that, as its currently allowed to have an empty `Root.fusion`

that would be breaking so im reverting this.

![image](https://user-images.githubusercontent.com/85400359/197333973-9408e771-ae5f-4478-a674-7858f29d563a.png)